File: lib/src/structuredfield/SfItem.ts
import type { SfBareItem } from './SfBareItem.js';
import type { SfParameters } from './SfParameters.js';

/**
 * Structured Field Item
 *
 * @group Structured Field
 *
 * @beta
 */
export class SfItem {

	value: SfBareItem;

	params?: SfParameters;

	constructor(value: any, params?: SfParameters) {
		if (Array.isArray(value)) {
			value = value.map((v) => (v instanceof SfItem) ? v : new SfItem(v));
		}

		this.value = value;
		this.params = params;
	}
}

File: python/src/structuredfield/SfItem.py
class SfItem:
    """
    Structured Field Item

          @group Structured Field

          @beta
    """

    def __init__(self, value, params=None):
        if type(value) == list:
            value = list(map(lambda v: v if type(v) == SfItem else SfItem(v), value))

        self.value = value
        self.params = params

    def __eq__(self, other):
        if not isinstance(other, SfItem):
            return False

        return self.value == other.value and self.params == other.params
